src/sentry/api/endpoints/organization_events_meta.py

```diff
@@ -120,63 +120,82 @@ def get(self, request: Request, organization) -> Response:
         except NoProjects:
             return Response({})
 
+        dataset_field_mapping = {
+            discover: {
+                "duration_column": "transaction.duration",
+                "id": "id",
+            },
+            spans_indexed: {
+                "duration_column": "span.self_time",
+                "id": "span_id",
+            },
+        }
+
+        dataset = self.get_dataset(request, "spansIndexed")
+        if dataset not in dataset_field_mapping.keys():
+            raise ParseError("Requested dataset is not supported")
+        dataset_fields = dataset_field_mapping[dataset]
+
         buckets = request.GET.get("intervals", 3)
-        lower_bound = request.GET.get("lowerBound", 0)
         first_bound = request.GET.get("firstBound")
         second_bound = request.GET.get("secondBound")
-        upper_bound = request.GET.get("upperBound")
-        column = request.GET.get("column", "span.self_time")
+        seed = request.GET.get("seed")
+
+        try:
+            lower_bound = int(request.GET.get("lowerBound", 0))
+            upper_bound = int(request.GET.get("upperBound"))
+        except ValueError:
+            raise ParseError("lower and upper bound both must be numbers")
```

Comment:
nit: since this could also be `upperBound` isn't provided, should we call it out as a required param